packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Shared/StringExtensions.cs

```diff
@@ -222,6 +222,22 @@ public static string FirstCharToUpperCase(this string str)
             return new string(span);
         }
 
+        public static string FirstCharToLowerCase(this string str)
+        {
+            if (string.IsNullOrEmpty(str))
+                return str;
+
+            var strSpan = str.AsSpan();
+
+            if (char.IsLower(strSpan[0]))
+                return str;
+
+            Span<char> span = stackalloc char[strSpan.Length];
+            strSpan.CopyTo(span);
```

Comment:
I believe this should have better performance or similar because `string.ToCharArray` copies everything to a new array as well.
Also this is not new code in this PR, I just copied everything in `FirstCharToLowerCase` method.

Reply:
yes, the performance should be the same, just saying the code could be more compact, it's a nit anyway.

Reply:
feel free to change it in another PR and maybe include a benchmark for it.
Since this PR is not introducing it, I would not change it.
I will consider to remove this method if it is unused.

Reply:
So it turns out this method is no longer used. I removed it.